completions/bash/multipass

```diff
@@ -320,22 +408,35 @@ _multipass_complete()
     if [[ "$prev_opts" = false ]]; then
         case "${cmd}" in
             "exec"|"stop"|"suspend"|"restart")
-                _multipass_instances "Running"
+                _multipass_instances_unique "Running"
             ;;
             "connect"|"sh"|"shell")
-                _multipass_instances "Running"
-                _multipass_instances "Stopped"
-                _multipass_instances "Suspended"
+                _multipass_instances_unique "Running"
+                _multipass_instances_unique "Stopped"
+                _multipass_instances_unique "Suspended"
             ;;
             "start")
-                _multipass_instances "Stopped"
-                _multipass_instances "Suspended"
+                _multipass_instances_unique "Stopped"
+                _multipass_instances_unique "Suspended"
             ;;
-            "delete"|"info"|"umount"|"unmount")
+            "umount"|"unmount")
                 _multipass_instances
             ;;
+            "delete"|"info")
+                _multipass_instances_and_snapshots
+            ;;
             "recover")
-                _multipass_instances "Deleted"
+                _multipass_instances_unique "Deleted"
+            ;;
+            "snapshot")
+                _multipass_instances_unique "Stopped"
+            ;;
+            "restore")
+                if [[ "${cur}" =~ '.' ]]; then
+                    _multipass_snapshots
+                else
+                    _multipass_restorable_instances
+                fi
```

Comment:
Again, I don't understand the two-step approach. Is this motivated by `_get_comp_words_by_ref` splitting on dots? You can avoid that by passing `-n '.'`. See `/usr/share/bash-completion/bash_completion`.